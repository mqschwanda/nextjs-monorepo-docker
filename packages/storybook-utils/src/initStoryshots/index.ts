import type { StoryshotsOptions } from '@storybook/addon-storyshots/dist/api/StoryshotsOptions';
import initAddonStoryshots from '@storybook/addon-storyshots';

/**
 * Options for the `initStoryshots` function.
 */
export type InitStoryshotsOptions = StoryshotsOptions & {
  /**
   * If the snapshot tests fail when react logs a key error.
   *
   * Set this to `false` if stories avoid using the key because it would show up in the code example.
   *
   * @default true
   */
  failOnReactKeyError?: boolean
};

/**
 * A wrapper for the `@storybook/addon-storyshots` module
 *
 * See [addon storyshots docs](https://storybook.js.org/addons/@storybook/addon-storyshots) for more information.
 */
export default function initStoryshots({
  failOnReactKeyError = true,
  ...options
}: InitStoryshotsOptions) {
  if (failOnReactKeyError === false) {
    const consoleError = jest.spyOn(console, 'error');

    beforeAll(() => {
      consoleError.mockImplementation((...args) => {
        const [message] = args;
        if (message.includes('Each child in a list should have a unique "key" prop.')) {
          return;
        }

        console.error(...args); // eslint-disable-line no-console
      });
    });

    afterAll(() => {
      consoleError.mockRestore();
    });
  }

  initAddonStoryshots(options);
}
