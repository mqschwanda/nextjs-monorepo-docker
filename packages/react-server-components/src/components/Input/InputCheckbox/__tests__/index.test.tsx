import { render } from '@testing-library/react';
import { InputCheckbox } from '@';
import initStoryshots from '@/__tests__/initStoryshots';

initStoryshots({
  failOnReactKeyError: false,
  storyKindRegex: /^@mqs\/react-server-components\/components\/Input\/InputCheckbox$/,
});

describe('@mqs/react-server-components', () => {
  describe('components', () => {
    describe('<InputCheckbox />', () => {
      it('renders', () => {
        const testId = InputCheckbox.name;

        const { getByTestId, asFragment } = render((
          <InputCheckbox
            className='test'
          />
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
