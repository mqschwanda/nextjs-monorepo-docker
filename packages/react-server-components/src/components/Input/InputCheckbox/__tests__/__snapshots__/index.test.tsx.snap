// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mqs/react-server-components components <InputCheckbox /> renders 1`] = `
<DocumentFragment>
  <input
    class="checkbox checkbox-primary checkbox-md test"
    data-testid="InputCheckbox"
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`Storyshots @mqs/react-server-components/components/Input/InputCheckbox Input Checkbox Color Example 1`] = `
[
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      accent
    </label>
    <input
      className="checkbox checkbox-accent checkbox-md"
      data-testid="InputCheckbox"
      type="checkbox"
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      error
    </label>
    <input
      className="checkbox checkbox-error checkbox-md"
      data-testid="InputCheckbox"
      type="checkbox"
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      info
    </label>
    <input
      className="checkbox checkbox-info checkbox-md"
      data-testid="InputCheckbox"
      type="checkbox"
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      primary
    </label>
    <input
      className="checkbox checkbox-primary checkbox-md"
      data-testid="InputCheckbox"
      type="checkbox"
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      secondary
    </label>
    <input
      className="checkbox checkbox-secondary checkbox-md"
      data-testid="InputCheckbox"
      type="checkbox"
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      success
    </label>
    <input
      className="checkbox checkbox-success checkbox-md"
      data-testid="InputCheckbox"
      type="checkbox"
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      warning
    </label>
    <input
      className="checkbox checkbox-warning checkbox-md"
      data-testid="InputCheckbox"
      type="checkbox"
    />
  </div>,
]
`;

exports[`Storyshots @mqs/react-server-components/components/Input/InputCheckbox Input Checkbox Example 1`] = `
<input
  className="checkbox checkbox-primary checkbox-md"
  data-testid="InputCheckbox"
  type="checkbox"
/>
`;

exports[`Storyshots @mqs/react-server-components/components/Input/InputCheckbox Input Checkbox Size Example 1`] = `
[
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      lg
    </label>
    <input
      className="checkbox checkbox-primary checkbox-lg"
      data-testid="InputCheckbox"
      type="checkbox"
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      md
    </label>
    <input
      className="checkbox checkbox-primary checkbox-md"
      data-testid="InputCheckbox"
      type="checkbox"
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      sm
    </label>
    <input
      className="checkbox checkbox-primary checkbox-sm"
      data-testid="InputCheckbox"
      type="checkbox"
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      xs
    </label>
    <input
      className="checkbox checkbox-primary checkbox-xs"
      data-testid="InputCheckbox"
      type="checkbox"
    />
  </div>,
]
`;
